control-plane/api-gateway/gatekeeper/gatekeeper_test.go

```diff
@@ -40,12 +40,19 @@ var (
 			Name:     "Listener 1",
 			Port:     8080,
 			Protocol: "TCP",
+			Hostname: common.PointerTo(gwv1beta1.Hostname("example.com")),
 		},
 		{
 			Name:     "Listener 2",
 			Port:     8081,
 			Protocol: "TCP",
 		},
+		{
+			Name:     "Listener 3",
+			Port:     8080,
+			Protocol: "TCP",
+			Hostname: common.PointerTo(gwv1beta1.Hostname("example.net")),
+		},
```

Comment:
We have multiple asserts on the `Service` being created. Adding a listener here with a duplicate port makes these tests more robust by ensuring our existing assertions pass.